src/app/store/machine/slice.ts

```diff
@@ -1682,6 +1696,300 @@ const machineSlice = createSlice({
+      },
+    },
+    getFilterOptionsStart: {
+      prepare: (action) => action,
+      reducer: (
+        state,
+        action: PayloadAction<null, string, { identifier: string }>
+      ) => {
+        const {
+          meta: { identifier },
+        } = action;
+        const group = state.filterGroups.find(
+          (group) => group.key === identifier
+        );
+        if (group) {
+          group.optionsLoaded = false;
+          group.optionsLoading = true;
+        }
+      },
+    },
+    getFilterOptionsSuccess: {
+      prepare: (action) => action,
+      reducer: (
+        state,
+        action: PayloadAction<
+          FilterOptionsResponse,
+          string,
+          { identifier: string }
+        >
+      ) => {
+        const {
+          meta: { identifier },
+          payload,
+        } = action;
+        const group = state.filterGroups.find(
+          (group) => group.key === identifier
+        );
+        if (group) {
+          group.options = payload;
+          group.optionsLoaded = true;
+          group.optionsLoading = false;
+        }
+      },
+    },
+    getFilterOptionsError: {
+      prepare: (action) => action,
+      reducer: (
+        state,
+        action: PayloadAction<APIError, string, { identifier: string }>
+      ) => {
+        const {
+          meta: { identifier },
+        } = action;
+        const group = state.filterGroups.find(
+          (group) => group.key === identifier
+        );
+        if (group) {
+          group.optionsLoaded = false;
+          group.optionsLoading = false;
+        }
+        state.errors = action.payload;
+      },
+    },
+    list: {
+      prepare: (listId: string, params: ListParams) => ({
+        meta: {
+          model: MachineMeta.MODEL,
+          method: "list",
+          nocache: true,
+          identifier: listId,
+        },
+        payload: {
+          params,
+        },
+      }),
+      reducer: () => {
+        // No state changes need to be handled for this action.
+      },
+    },
+    listStart: {
+      prepare: (action) => action,
+      reducer: (
+        state,
+        action: PayloadAction<null, string, { identifier: string }>
+      ) => {
+        const {
+          meta: { identifier },
+        } = action;
+        state.lists[identifier] = {
+          count: 0,
+          cur_page: 1,
+          errors: null,
+          groups: [],
+          loaded: false,
+          loading: true,
+          num_pages: 1,
+        };
+      },
+    },
+    listSuccess: {
+      prepare: (action) => action,
+      reducer: (
+        state,
+        action: PayloadAction<ListResponse, string, { identifier: string }>
+      ) => {
+        const {
+          meta: { identifier },
+          payload,
+        } = action;
+        const list = state.lists[identifier];
+
+        if (!list) {
+          return;
+        }
+
+        // Convert ungrouped response into null group
```

Comment:
The API payload has a different shape depending on whether `group_key` was included in the request, so it's normalised in the reducer.